sky/clouds/service_catalog/common.py

```diff
@@ -55,23 +59,52 @@ def read_catalog(filename: str) -> pd.DataFrame:
     assert filename.endswith('.csv'), 'The catalog file must be a CSV file.'
     catalog_path = get_catalog_path(filename)
     cloud = cloud_lib.CLOUD_REGISTRY.from_str(os.path.dirname(filename))
-    if not os.path.exists(catalog_path):
-        url = f'{constants.HOSTED_CATALOG_DIR_URL}/{constants.CATALOG_SCHEMA_VERSION}/{filename}'  # pylint: disable=line-too-long
-        with backend_utils.safe_console_status(
-                f'Downloading {cloud} catalog...'):
-            try:
-                r = requests.get(url)
-                r.raise_for_status()
-            except requests.exceptions.RequestException as e:
-                logger.error(f'Failed to download {cloud} catalog:')
-                with ux_utils.print_exception_no_traceback():
-                    raise e
-        # Save the catalog to a local file.
-        os.makedirs(os.path.dirname(catalog_path), exist_ok=True)
-        with open(catalog_path, 'w') as f:
-            f.write(r.text)
-        logger.info(f'A new {cloud} catalog has been downloaded to '
-                    f'{catalog_path}')
+
+    meta_path = os.path.join(_CATALOG_DIR, '.meta', filename)
+    os.makedirs(os.path.dirname(meta_path), exist_ok=True)
+
+    def _need_update() -> bool:
```

Comment:
Consider adding a comment `# Must be called under a FileLock below.` or move L88 above here. (Was a bit surprised.)